src/BendersBase/test/test_autodecompose.jl

```diff
@@ -0,0 +1,147 @@
+using Test
+using BendersBase
+using JuMP
+
+@testset "Auto Decompose" begin
+
+    @testset "Basic Functionality" begin
+        @testset "Simple facility location model" begin
+            # Build a simple facility location model
+            model = Model()
+            @variable(model, x[1:3], Bin)
+            @variable(model, y[1:3, 1:2] >= 0)
+            @objective(model, Min, 10*x[1] + 15*x[2] + 12*x[3] + sum(y))
+            @constraint(model, sum(x) >= 1)
+            @constraint(model, demand[j in 1:2], sum(y[:,j]) == 1)
+            @constraint(model, capacity[i in 1:3, j in 1:2], y[i,j] <= x[i])
+
+            # Decompose the model
+            data, master, oracle = auto_decompose(model)
+
+            # Verify Data structure
+            @test data.dim_x == 3
+            @test data.dim_t == 1
+            @test length(data.c_x) == 3
+            @test length(data.c_t) == 1
+            @test data.c_t[1] == 1.0
+
+            # Verify Master model structure
+            @test num_variables(master.model) == 4  # 3 x variables + 1 t variable
+            @test length(master.x) == 3
+            @test length(master.t) == 1
+
+            # Verify Oracle model structure
+            @test length(oracle.fixing_constraints) == 3  # One fixing constraint per master variable
+        end
+    end
+
+    @testset "Classification Tests" begin
+        @testset "Constraint partitioning" begin
+            # Build model
+            model = Model()
+            @variable(model, x[1:3], Bin)
+            @variable(model, y[1:3, 1:2] >= 0)
+            @objective(model, Min, 10*x[1] + 15*x[2] + 12*x[3] + sum(y))
+            @constraint(model, sum(x) >= 1)
+            @constraint(model, demand[j in 1:2], sum(y[:,j]) == 1)
+            @constraint(model, capacity[i in 1:3, j in 1:2], y[i,j] <= x[i])
+
+            # Classify variables
+            master_vars, oracle_vars = BendersBase.classify_variables_for_benders(model, Symbol[])
```

Comment:
Please add a testset for an example with fractional master variables to test `decision_var` in `classify_variables_for_benders`.